### PR TITLE
Nextcloud: API invite manager

### PR DESCRIPTION
**Pull request by parhamin2010**
- Create new database tables
- Create new routes
- Define ApiController and six method for now as endpoint

### CODE REVIEW

File: appinfo/routes.php [hunk 1]
```diff
@@ -89,8 +89,16 @@
 		["name" => "settings#get_settings", "url" => "/ajax/settings", "verb" => "GET"],
 		["name" => "settings#get_sciencemesh_settings", "url" => "/sciencemesh_settings", "verb" => "GET"],
 		["name" => "settings#save_sciencemesh_settings", "url" => "/ajax/sciencemesh_settings/save", "verb" => "GET"],
-		["name" => "settings#check_connection_settings", "url" => "/ajax/check_connection_settings", "verb" => "GET"]
-
+		["name" => "settings#check_connection_settings", "url" => "/ajax/check_connection_settings", "verb" => "GET"],
+		
+		// API Routes
+		["name" => "Api#add_token", "url" => "/api/v1/add_token/{initiator}", "verb" => "POST"],
+		["name" => "Api#get_token", "url" => "/api/v1/get_token/{initiator}", "verb" => "GET"],
```
Comment (mrvahedi68) [inline appinfo/routes.php:96]: From Reva, no initiator exists for this route. 

["name" => "Api#get_token", "url" => "/api/v1/get_token", "verb" => "GET"],

File: lib/Controller/ApiController.php [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+			$qb->expr()->eq('initiator', $qb->createNamedParameter($initiator, IQueryBuilder::PARAM_STR))
+		)
+		->andWhere(
+			$qb->expr()->eq('idp', $qb->createNamedParameter($this->request->getParam('idp'), IQueryBuilder::PARAM_STR))
+		)
+		->andWhere(
+			$qb->expr()->eq('opaque_user_id', $qb->createNamedParameter($this->request->getParam('opaqueUserId'), IQueryBuilder::PARAM_STR))
+		)
+		->andWhere(
+			$qb->expr()->eq('email', $qb->createNamedParameter($this->request->getParam('email'), IQueryBuilder::PARAM_STR))
```
Comment (mrvahedi68) [inline lib/Controller/ApiController.php:272]: Not exists in the input list. only `userId`, `idp` will pass to the API.

**Review by mrvahedi68 — CHANGES_REQUESTED** [review]

File: appinfo/routes.php (hunk 1 above)
Comment (parhamin2010) [reply appinfo/routes.php:96]: fixed

File: lib/Controller/ApiController.php (hunk 2 above)
Comment (parhamin2010) [reply lib/Controller/ApiController.php:269]: fixed

**Comment by parhamin2010** [comment]
@mrvahedi68 plz double-review it and merge if everythings good.

**Pull request closed without merging**